templates/repo/issue/view_content/comments.tmpl

```diff
@@ -707,6 +707,11 @@
 						{{$.locale.TrN (len .Commits) "repo.issues.push_commit_1" "repo.issues.push_commits_n" (len .Commits) $createdStr | Safe}}
 					{{end}}
 				</span>
+				{{if and .IsForcePush $.Issue.PullRequest.BaseRepo.Name}}
+				<span class="ui float right comparebox">
+					<a href="{{$.Issue.PullRequest.BaseRepo.Link}}/compare/{{PathEscape .OldCommit}}..{{PathEscape .NewCommit}}" rel="nofollow" class="ui compare label">{{$.locale.Tr "repo.issues.force_push_compare"}}</a>
```

Comment:
Ah, I already wondered how you would solve the problem that a force-pushed commit might have been deleted by Git GC. Your approach works of course, as it will simply display a `404` or similar…

---

I'm wondering: Should we perhaps open the comparison in a new tab?
You probably want to continue using this tab, if you're already taking the time to go through the individual commit history…